platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/GossipEvent.java

```diff
@@ -32,13 +35,19 @@
  */
 public class GossipEvent implements EventIntakeTask, BaseEvent, ChatterEvent {
     private static final long CLASS_ID = 0xfe16b46795bfb8dcL;
+
     private static final long ROUND_CREATED_UNDEFINED = -1;
     private BaseEventHashedData hashedData;
     private BaseEventUnhashedData unhashedData;
     private EventDescriptor descriptor;
     private Instant timeReceived;
     private long roundCreated = ROUND_CREATED_UNDEFINED;
 
+    /**
+     * The id of the node which sent us this event
+     */
+    private NodeId senderId;
```

Comment:
I'd recommend adding the following to the javadoc (just in case the person who eventually creates the protobuf is tempted to include this field).

```
The sender ID of an event should not be serialized when an event is serialized, and it should not effect the hash of the event in any way.
```

Reply:
added